src/compiler/checker.ts

```diff
@@ -14820,7 +14813,7 @@ namespace ts {
                 // This will allow excess properties in spread type as it is very common pattern to spread outter attributes into React component in its render method.
                 if (isSourceAttributeTypeAssignableToTarget && !isTypeAny(sourceAttributesType) && !isTypeAny(targetAttributesType)) {
                     for (const attribute of openingLikeElement.attributes.properties) {
-                        if (isJsxAttribute(attribute) && !isKnownProperty(targetAttributesType, attribute.name.escapedText, /*isComparingJsxAttributes*/ true)) {
+                        if (isJsxAttribute(attribute) && (isUnhyphenatedJsxName(idText(attribute.name)) || !!(getPropertyOfType(targetAttributesType, attribute.name.escapedText))) && !isKnownProperty(targetAttributesType, attribute.name.escapedText, /*isComparingJsxAttributes*/ true)) {

```

Comment:
this line is too long

Reply:
also, I thought we had a standalone predicate for this. isIgnoredJsxProperty?

Reply:
`isIgnoredJsxProperty` works on the source type, source prop symbol, and target type (optional) - this uses the attribute name and symbol presence instead. So, I can have this
```ts
 const isNotIgnoredJsxProperty = (isUnhyphenatedJsxName(idText(attrName)) || !!(getPropertyOfType(targetAttributesType, attrName.escapedText)));
```
or this:
```ts
 const isNotIgnoredJsxProperty = !isIgnoredJsxProperty(sourceAttributesType, getPropertyOfObjectType(sourceAttributesType, attrName.escapedText), getTypeOfPropertyOfType(targetAttributesType, attrName.escapedText));
```